File: y.py

'''import requests

# Replace this with the URL of your PHP file on GitHub
github_php_url = "https://github1s.com/JayPatel-2023/SQL-OnlineDB/blob/main/y.php"

# Make an HTTP request to the PHP file
response = requests.get(github_php_url)

# Check if the request was successful (status code 200)
if response.status_code == 200:
    # Print the content returned by the PHP file
    print("Output from PHP:", response.text)
else:
    print(f"Error: {response.status_code}")
'''
import requests

# Replace this with the URL of your modified PHP file on GitHub
github_php_url = "https://github.com/JayPatel-2023/SQL-OnlineDB/blob/main/y.php"

# Specify the function to call
function_name = "getData"

# Make an HTTP request to the PHP file with the function parameter
response = requests.get(github_php_url, params={"function": function_name})

# Check if the request was successful (status code 200)
if response.status_code == 200:
    # Parse the JSON response
    result = response.json()
    print("Result from PHP function:", result)
else:
    print(f"Error: {response.status_code}")

'''
import requests
import json

# Replace this with the URL of your modified PHP file on GitHub
github_php_url = "https://github.com/JayPatel-2023/SQL-OnlineDB/blob/main/y.php"

# Specify the function to call
function_name = "getData"

# Make an HTTP request to the PHP file with the function parameter
response = requests.get(github_php_url, params={"function": function_name})

# Check if the request was successful (status code 200)
if response.status_code == 200:
    # Parse the JSON response
    result = response.json()

    # Print the specific data from the function
    if 'message' in result:
        print(result['message'])
    else:
        print("Unexpected JSON format")
else:
    print(f"Error: {response.status_code}")
'''

File: y.php
<?php
// example.php

function getData() {
    // Your existing PHP code to fetch data from the database
    // ...
    $result = 'jay';
    // Return the data as an associative array
    $data = array(
        'message' => 'Hello from getData function!',
        'result' => $result
    );

    // Convert the array to JSON and echo it
    echo json_encode($data);
}

function anotherFunction() {
    // Another function in your PHP code
    // ...
    $data = 'jay';
    // Return the result as an associative array
    $result = array(
        'message' => 'Hello from anotherFunction!',
        'data' => $data
    );

    // Convert the array to JSON and echo it
    echo json_encode($result);
}

// Check if a function parameter is provided in the URL
if (isset($_GET['function'])) {
    $functionName = $_GET['function'];

    // Call the specified function
    if (function_exists($functionName)) {
        // Disable HTML rendering
        header('Content-Type: application/json');

        // Call the specified function
        call_user_func($functionName);
    } else {
        // Function not found
        echo json_encode(["error" => "Function not found"]);
    }
} else {
    // Function parameter missing
    echo json_encode(["error" => "Function parameter missing"]);
}
?>
